pkg/healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,355 @@
+			_, err := LoadConfig("")
+			Expect(err).To(HaveOccurred())
+			t.Setenv(CONFIG_ENV, "")
+		})
+		var _ = It("return error if config is invalid", func() {
+			filePath, err := os.CreateTemp(tmpPath, "healtchecktest")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = filePath.WriteString(fakeConfInvalid)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = LoadConfig(filePath.Name())
+			Expect(err).To(HaveOccurred())
+		})
+		var _ = It("return no error if valid config exists", func() {
+			filePath, err := os.CreateTemp(tmpPath, "healtchecktest")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = filePath.WriteString(fakeConf)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = LoadConfig(filePath.Name())
+			Expect(err).ToNot(HaveOccurred())
+		})
+		var _ = It("return no error if config does not exist but is not mandatory", func() {
+			conf, err := LoadConfig("/some/invalid/path")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(conf).ToNot(BeNil())
+		})
+	})
+	var _ = Context("IsFRRActive() should", func() {
+		var _ = It("return error if FRR Manager returns error", func() {
+			c := fake.NewSimpleClientset()
+			frm := &fakeFRRManagerError{}
+			nc := &NetHealthcheckConfig{}
+			hc, err := NewHealthChecker(c, NewDefaultHealthcheckToolkit(frm, NewTcpDialer("")), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			result, err := hc.IsFRRActive()
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeFalse())
+		})
+		var _ = It("return error if FRR is inactive", func() {
+			c := fake.NewSimpleClientset()
+			frm := &fakeFRRManagerInactive{}
+			nc := &NetHealthcheckConfig{}
+			hc, err := NewHealthChecker(c, NewDefaultHealthcheckToolkit(frm, NewTcpDialer("")), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			result, err := hc.IsFRRActive()
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeFalse())
+		})
+		var _ = It("return no error if FRR is active", func() {
+			c := fake.NewSimpleClientset()
+			frm := &fakeFRRManagerSuccess{}
+			nc := &NetHealthcheckConfig{}
+			hc, err := NewHealthChecker(c, NewDefaultHealthcheckToolkit(frm, NewTcpDialer("")), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			result, err := hc.IsFRRActive()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(BeTrue())
+		})
+	})
+	var _ = Context("RemoveTaint() should", func() {
+		var _ = It("return error about no nodes", func() {
+			c := fake.NewSimpleClientset()
+			nc := &NetHealthcheckConfig{}
+			hc, err := NewHealthChecker(c, nil, nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			err = hc.RemoveTaint(TAINT_KEY)
+			Expect(err).To(HaveOccurred())
+			Expect(hc.IsInitialized()).To(BeFalse())
+		})
+		var _ = It("return error when trying to remove taint (update node)", func() {
+			c := fake.NewSimpleClientset(fakeNodes)
+			c.CoreV1().(*fakecorev1.FakeCoreV1).PrependReactor("update", "nodes",
```

Comment:
controller-runtime also has a pretty nice fakeclient if you are able to use that one

Reply:
Done

Although, had to add a mock for this client's error, as can't find anything similar to `PrependReactor` available in goclient.